examples/java/src/main/java/glide/examples/ClusterExample.java

```diff
@@ -0,0 +1,152 @@
+        }
+        return null;
+    }
+
+    /**
+     * Executes the main logic of the application, performing basic operations such as SET, GET, PING,
+     * and INFO REPLICATION using the provided GlideClusterClient.
+     *
+     * @param client An instance of <code>GlideClusterClient</code>.
+     */
+    public static void appLogic(GlideClusterClient client) {
+
+        // Send SET and GET
+        try {
+            CompletableFuture<String> setResponse = client.set("foo", "bar");
+            Logger.log(Logger.Level.INFO, "app", "Set response is " + setResponse.get());
+        } catch (ExecutionException | InterruptedException e) {
+            Logger.log(Logger.Level.ERROR, "glide", "Error during SET: " + e.getMessage());
+        }
+
+        try {
+            CompletableFuture<String> getResponse = client.get("foo");
+            Logger.log(Logger.Level.INFO, "app", "Get response is " + getResponse.get());
+        } catch (ExecutionException | InterruptedException e) {
+            Logger.log(Logger.Level.ERROR, "glide", "Error during GET: " + e.getMessage());
+        }
+
+        // Send PING to all primaries (according to Valkey's PING request_policy)
+        try {
+            CompletableFuture<String> pong = client.ping();
+            Logger.log(Logger.Level.INFO, "app", "Ping response is " + pong.get());
+        } catch (ExecutionException | InterruptedException e) {
+            Logger.log(Logger.Level.ERROR, "glide", "Error during PING: " + e.getMessage());
+        }
+
+        // Send INFO REPLICATION with routing option to all nodes
+        try {
+            ClusterValue<String> infoResponse =
+                    client.info(
+                            InfoOptions.builder().section(InfoOptions.Section.REPLICATION).build(), ALL_NODES).get();
+            Logger.log(
+                    Logger.Level.INFO,
+                    "app",
+                    "INFO REPLICATION responses from all nodes are " + infoResponse.getMultiValue());
+        } catch (ExecutionException | InterruptedException e) {
+            Logger.log(Logger.Level.ERROR, "glide", "Error during INFO: " + e.getMessage());
+        }
+    }
+
+    /** Executes the application logic with exception handling. */
+    private static void execAppLogic() {
+
+        while (true) {
+            GlideClusterClient client = null;
+            try {
+                client = createClient();
+
+                assert client != null;
+                appLogic(client);
+                return;
+
+            } catch (ClosingException e) {
+                // If the error message contains "NOAUTH", raise the exception
+                // because it indicates a critical authentication issue.
+                if (e.getMessage().contains("NOAUTH")) {
+                    Logger.log(
+                            Logger.Level.ERROR, "glide", "Authentication error encountered: " + e.getMessage());
```

Comment:
The comment mentions we should raise the exception but we don't do that here, should we add a throw here?
```suggestion
                            Logger.Level.ERROR, "glide", "Authentication error encountered: " + e.getMessage());
                            throw e;
```